mutatex/utils.py

```diff
@@ -421,43 +422,59 @@ def get_foldx_sequence(pdb, multimers=True):
 
     residue_list = []
     sequences = {}
+    positions = {}
+
     for model in structure:
         for chain in model:
             chain_name = chain.get_id()
             sequences[chain_name] = ''
+            positions[chain_name] = ''
             for residue in chain:
                 try:
                     res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
+                    pos_code = str(residue.get_id()[1])
                 except:
                     log.warning("Residue %s in file %s couldn't be recognized; it will be skipped" %(residue, pdb))
                     continue
                 if not multimers:
                     residue_list.append(tuple(["%s%s%d" % (res_code, chain.get_id(), residue.get_id()[1])]))
                 else:
                     sequences[chain_name] += res_code
+                    positions[chain_name] += pos_code
 
     if multimers:
-        collated_chains = []
+        collated_chains_pos = []
+        collated_chains_seq = []
         seq_ids, seqs = list(zip(*list(iteritems(sequences))))
+        pos_ids, pos = list(zip(*list(iteritems(positions))))
         seq_ids = np.array(seq_ids)
+        pos_ids = np.array(pos_ids)
+
         unique_seqs, unique_idxs = np.unique(seqs, return_inverse=True)
+        unique_pos, unique_idxp = np.unique(pos, return_inverse=True)
 
+        for i in np.unique(unique_idxp):
+            collated_chains_pos.append(pos_ids[unique_idxp == i])
         for i in np.unique(unique_idxs):
-            collated_chains.append(seq_ids[unique_idxs == i])
+            collated_chains_seq.append(seq_ids[unique_idxs == i])
+
+        if len(collated_chains_seq) > len(collated_chains_pos) or len(collated_chains_seq) == len(collated_chains_pos):
```

Comment:
I don't think checking the length guarantees that pairs of chains have the same numbering. Here is an example:

```
seq A:
A C D E
1 2 3 4

seq B:
A C D E
5 6 7 8

seq C:
F G H I
5 6 7 8
```

In this case I think you would have `[[A, B], [C]]` for the sequence and `[[A], [B, C]]` for the numbering and they have the same length. Instead of checking for length, you can just check the numbering, i.e. do sequences that are in the same group have identical numbering?